nautobot/core/tests/test_jobs.py

```diff
@@ -606,3 +608,110 @@ def test_cleanup_object_changes(self):
         self.assertTrue(JobResult.objects.filter(date_done__gte=cutoff).exists())
         self.assertFalse(ObjectChange.objects.filter(time__lt=cutoff).exists())
         self.assertTrue(ObjectChange.objects.filter(time__gte=cutoff).exists())
+
+
+class BulkDeleteTestCase(TransactionTestCase):
+    """
+    Test the BulkDelete system job.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.status_ct = ContentType.objects.get_for_model(Status)
+        self.role_ct = ContentType.objects.get_for_model(Role)
+        self.device_ct = ContentType.objects.get_for_model(Device)
+
+    def _common_no_error_test_assertion(self, model, job_result, **filter_params):
+        self.assertEqual(job_result.status, JobResultStatusChoices.STATUS_SUCCESS)
+        self.assertEqual(model.objects.filter(**filter_params).count(), 0)
+        self.assertFalse(
+            JobLogEntry.objects.filter(job_result=job_result, log_level=LogLevelChoices.LOG_WARNING).exists()
+        )
+        self.assertFalse(
+            JobLogEntry.objects.filter(job_result=job_result, log_level=LogLevelChoices.LOG_ERROR).exists()
+        )
+
+    def test_bulk_delete_without_permission(self):
+        statuses_to_delete = [str(status) for status in Status.objects.all().values_list("pk", flat=True)[:2]]
+        job_result = create_job_result_and_run_job(
+            "nautobot.core.jobs.bulk_actions",
+            "BulkDeleteObjects",
+            content_type=self.status_ct.id,
+            pk_list=statuses_to_delete,
+            username=self.user.username,
+        )
+        self.assertEqual(job_result.status, JobResultStatusChoices.STATUS_FAILURE)
+        job_log = JobLogEntry.objects.get(job_result=job_result, log_level=LogLevelChoices.LOG_ERROR)
+        self.assertEqual(job_log.message, f'User "{self.user}" does not have permission to delete status objects')
+
+    def test_bulk_delete_objects_with_constrained_permission(self):
+        statuses_to_delete = [str(status) for status in Status.objects.all().values_list("pk", flat=True)[:2]]
+        obj_perm = ObjectPermission(
+            name="Test permission",
+            constraints={"pk": str(uuid.uuid4())},  # Match a non-existent pk (i.e., deny all)
+            actions=["delete"],
+        )
+        obj_perm.save()
+        obj_perm.users.add(self.user)
+        obj_perm.object_types.add(ContentType.objects.get_for_model(Status))
+
+        job_result = create_job_result_and_run_job(
+            "nautobot.core.jobs.bulk_actions",
+            "BulkDeleteObjects",
+            content_type=self.status_ct.id,
+            pk_list=statuses_to_delete,
+            username=self.user.username,
+        )
+        self.assertEqual(job_result.status, JobResultStatusChoices.STATUS_FAILURE)
+        job_logs = JobLogEntry.objects.filter(job_result=job_result, log_level=LogLevelChoices.LOG_INFO)
+        self.assertEqual(job_logs[1].message, "Deleting 0 statuses...")
```

Comment:
Feels a bit weird to me that we'd log this message instead of failing if no records are to be deleted? 

Though I see above that we expect this job to fail - but don't have any assertion about the nature of the failure message?